upgrade our postgrest

vwwv opened the pull request.

steve-chavez committed:
Add response.headers on GET/POST/PATCH/PUT/DELETE

steve-chavez committed:
Add request method and path GUCs

steve-chavez committed:
Add GUC for accessing the Authorization header

steve-chavez committed:
Allow overriding headers Location,Content-Type,etc

steve-chavez committed:
Update BACKERS.md

steve-chavez committed:
Change server-proxy-uri 2 openapi-server-proxy-uri

The server-proxy-uri config option mislead into thinking it was a
web server global setting when it's only related to OpenAPI.

steve-chavez committed:
Fix for sylish-haskell

robx committed:
Relax upper bounds for GHC 8.8.3 / stackage lts-15.3.

Specifically, allow base 4.13 which ships with GHC 8.8, as well
as several dependency updates including hasql-transaction 1.0,
swagger 2.5 and warp 3.3.

robx committed:
Raise lower bound for configurator-pg.

Building on GHC 8.8.3 requires the MonadFail instance that 0.1.0.6
provides.

robx committed:
Update stack.yaml.lock for HSYaml extra-deps.

robx committed:
Upgrade hspec-wai dependency to 0.10.

robx committed:
Fix test typo.

robx committed:
Update configurator-pg to 0.2.0 (better error messages) (#1457)

* Don't indent first line of configuration parser error messages

configurator-pg returns multi-line errors now. Alternatively,
we could indent all the lines.

steve-chavez committed:
Fix empty headers being added on POST/PATCH/DELETE (#1458)

mrexodia committed:
Improve command line help (#1465)

Make jwt-secret explicit about 32 characters minimum.

wolfgangwalther committed:
Add docker-compose file to run tests (#1463)

steve-chavez committed:
Allow multiple schemas to be exposed in one instance (#1450)

The schema to use can be selected through the headers `Accept-Profile` for GET/HEAD and `Content-Profile` for POST/PATCH/PUT/DELETE.

This is based on the https://www.w3.org/TR/dx-prof-conneg/ttps://www.w3.org/TR/dx-prof-conneg/ spec.

Also increase all memory tests by 1M(otherwise CI fails).

Co-authored-by: Mahmoud Kassem <MKassem@gk-software.com>
Co-authored-by: Mahmoud Kassem <mahmoud_k@mail.com>

steve-chavez committed:
Change docker image to support scram-sha-256 auth (#1467)

Debian stretch has a libpq < 10: https://packages.debian.org/stretch/libpq5.
But buster has libpq >= 10 https://packages.debian.org/buster/libpq5.

Fixes https://github.com/PostgREST/postgrest/issues/1443.

steve-chavez committed:
Bump to version v7.0.0

steve-chavez committed:
Fix circleci golang docker image

Upgrade golang otherwise ghr throws:
unrecognized import path "math/bits"

steve-chavez committed:
Fix compiling on windows(unix socket issue)

Group unix socket functionality into its own module

wolfgangwalther committed:
add postgres-12 tests to circleci (#1472)

* circleci: new postgres docker containers need host auth trust

wolfgangwalther committed:
Fix overloaded computed columns on RPC (#1473)

* fix some typos and spelling

* fix pg_source CTE name should be prefixed with pgrst_

* added tests for overloaded computed columns on patch calls

steve-chavez committed:
Update BACKERS.md

wolfgangwalther committed:
Fix POST, PATCH, DELETE with ?select= and empty body or return=minimal (#1471)

* add tests for POST, PATCH, DELETE when using ?select= with empty bodies or ret=min

* fix select with empty bodies and ret=min

* improved tests for default cases, fixed computed overloaded columns on empty-body?select= PATCH

monacoremo committed:
Fix CI tests for Postgres 9.4 (#1482)

monacoremo committed:
Add a script for running the test suite in a local temporary database. (#1476)

robx committed:
Update stack.yaml to lts-15.8 (GHC 8.8.3)

robx committed:
Remove unused imports (with newer protolude)

The changes here are from fixing stack build with LTS 15.8 (GHC 8.8.3)
to be warning-free. Most if not all come from the newer protolude
version, which adds some things like `unwords`.

So to make sure that we don't get unused import warnings with newer
stackage, this both raises the protolude lower bound and removes
these imports.

robx committed:
Relax upper bounds for lens, network-uri, time.

This allows lens-4.19, network-uri-2.7 and time-0.10. According to
the respective changelogs, none of these should cause trouble.

lens-4.19 is required to build with GHC 8.10.

robx committed:
Fix tests for explicit errors in aeson-1.4.7, and raise lower bound

robx committed:
Fix hlint 2.2.11 bracketing suggestions

robx committed:
circleci: Split style and hlint out into separate job

robx committed:
circleci: Break out a plain build and cache job

robx committed:
circleci: Dedicated cache key for style checks

robx committed:
circleci: Build linters with memory-saving ghc-options. (#1490)

Previously, the linters would use the cached build artefacts from
building postgrest with -fno-spec-constr via stack.yaml. Potentially
passing it to the linter build explicitly will fix recent out-of-memory
issues.

Commit:
Integrate stack in the nix-shell environment

monacoremo committed:
Basics for building PostgREST with Nix (#1489)

* Add GHC options used with stack to postgrest.cabal

* Add a Nix derivation for just the executable.

monacoremo committed:
Fix and improvements to with_tmp_db (#1488)

* Load fixtures in one psql session

* Add an option to keet the tmp directory and logs.

robx committed:
Update to protolude 0.3.0

The good part is that protolude 0.3.0 builds with GHC 8.10.

The bad part is that this change is a bit painful:
- the default `toS` has changed to no longer convert to and from ByteString
- similarly, `show` no longer outputs ByteString

The changes here are pretty much minimal to keep things compiling; I didn't
see a nice way to work with the new ConvertText class, even though `toUtf8`
seems like it might help if used besides `toS` at just the right spots.

monacoremo committed:
Disable nix pure by default in stack.yaml

monacoremo committed:
Remove ghc options from stack.yaml

They are redundant after being added to postgrest.cabal in #1489

monacoremo committed:
Run tests against several PostgreSQL versions using Nix (#1492)

* Show test output with postgrest-test-spec* in nix-shell

* Test output in color

monacoremo committed:
Pin version 0.3 of protolude in nix (#1499)

monacoremo committed:
Basic CI integration for Nix builds (#1495)

* Skip pushing to cachix for pull requests from forks.
* Build PostgREST first in the nix CI job in order to fail fast

robx committed:
Relax dependencies to build with GHC 8.10 (base, swagger2) (#1502)

* Allow swagger 2.7.
* Allow base-4.14 (GHC 8.10).

monacoremo committed:
Run the spec test suite from the nix build job (#1503)

* Add caching to the tests run in CI with Nix

monacoremo committed:
Add scripts for linting and styling with all dependencies managed by Nix (#1501)

* Integrate the style and lint dependencies with Nix

* Add documentation on the nix styling scripts.

robx committed:
Allow base64-bytestring 1.2.

steve-chavez committed:
Readd openapi-server-proxy-uri config option

This option accidentaly got lost during a merge.

Fixes https://github.com/PostgREST/postgrest/issues/1500

monacoremo committed:
Add tips and tricks to using nix-shell (#1505)

monacoremo committed:
Fix the cachix push logic in CircleCI

Before pushing to cachix, check if the CACHIX_SIGNING_KEY env var is available.

monacoremo committed:
Hook the Nix styling scripts into CI (#1504)

monacoremo committed:
More consistent loading of the test fixtures (#1506)

Remove db setup from the Haskell test suite and include it in the test bash scripts.

monacoremo committed:
Simplify the DbStructure queries (#1478)

* remove unused parts of the DbStructure queries

* explain magic numbers in DbStructure queries

* simplify primary keys query

steve-chavez committed:
Fix Content-Profile not working for POST RPC

Accept-Profile was applied instead.

Fixes https://github.com/PostgREST/postgrest/issues/1508.

steve-chavez committed:
Remove PUT restriction for all columns/single row

Fixes https://github.com/PostgREST/postgrest/issues/1452.

The single row restriction can be lifted because the
PUT will consider only the first object of the array.

steve-chavez committed:
Update BACKERS.md

steve-chavez committed:
appveyor: specify windows image

steve-chavez committed:
Add FreeBSD CI

monacoremo committed:
Add static build with Nix (fixes #1295) (#1494)

* Include modifications to nixpkgs and static-haskell-nix as patches.

* use the patched static-haskell-nix version

* Add cachix postgrest

steve-chavez committed:
circleci: add static bin to release

monacoremo committed:
Build Docker image from Nix (#1522)

* Static docker image

* added tags for latest and with version

* add postgrest repository to docker image name

* set creation date on nix docker images

steve-chavez committed:
cirrusci: add binary artifact

steve-chavez committed:
circleci: remove linux distro releases

Prefer the linux static binary for x64

steve-chavez committed:
Bump to version v7.0.1

steve-chavez committed:
Fix Dockerfile

Remove libpq5 dependency since static bin already contains it.

steve-chavez committed:
cabal: fix upload to hackage (#1532)

When doing:

```
cabal upload dist-newstyle/sdist/postgrest-7.0.1.tar.gz
```

The following error is shown:

```
Error: Invalid package

'ghc-options: -Wall -Werror' makes the package very easy to break with
future GHC versions because new GHC versions often add new warnings.
Use just 'ghc-options: -Wall' instead. Alternatively, if you want to
use this, make it conditional based on a Cabal configuration flag
(with 'manual: True' and 'default: False') and enable that flag
during development.
```

Put -Werror in a cabal flag to work around this restriction.

monacoremo committed:
Reduce size of the static docker image by patching the Nix openssl package (#1528)

* hook docker images into the default builds

monacoremo committed:
Fix UNKNOWN being shown as git commit hash (#1533)

Fixes https://github.com/PostgREST/postgrest/issues/1530

monacoremo committed:
Bump nixpkgs and simplify/clean up nix setup (#1529)

* Refactor tests and fix README

* fix linting hints from new hlint version

* postgrest-style with new nixpkgs-fmt

monacoremo committed:
Fix the static build (#1534)

* openssl linking fix is required after all

* remove Cabal override (obsoleted on new nixpkgs)

steve-chavez committed:
Add notice about single Linux static executable

monacoremo committed:
Nixify CircleCI setup (#1535)

* package ghr
* add nix-based release scripts
* refactor CI based on nix tests and release scripts
* update stack version in circleci
* update makefile
* remove docker directory
* add Docker Hub description
* add README on docker

felixonmars committed:
Allow aeson 1.5 (#1537)

monacoremo committed:
Nixify io and memory tests (#1538)

* Include tests for io and memory in the Nix environment.
* include spec tests in nix-shell by default
* install the io and memory tests in CI

steve-chavez committed:
Allow http status override through response.status guc (#1541)

Fixes https://github.com/PostgREST/postgrest/issues/1525

steve-chavez committed:
Fix broken links in .github dir

Also add a PR template

steve-chavez committed:
github: CHANGELOG reminder in PR template

steve-chavez committed:
refactor: separate reading file from parsing it

steve-chavez committed:
Allow schema cache reloading with NOTIFY (#1542)

Fixes https://github.com/PostgREST/postgrest/issues/1512

Helps on environments where you can't send unix signals(Windows, managed
containers). Also provides better UX for schema reloads - NOTIFY
can be sent from pg clients(psql, pgadmin).

`NOTIFY pgrst` - with no payload - should be done to reload the schema cache.
Notifications with a payload will be ignored.

The channel can be enabled with `db-channel-enabled`(false by default)
and its name can be configured with `db-channel`.

The LISTEN thread uses a dedicated pg connection.
This connection is recovered if it fails.
A debounce of 1ms is done in case too many NOTIFYs arrive.

steve-chavez committed:
Fix expired JWTs starting an empty transaction

Fixes https://github.com/PostgREST/postgrest/issues/1094.

Expired JWTs were doing an empty BEGIN/COMMIT in the db.

wolfgangwalther committed:
add test cases for singular minimal

wolfgangwalther committed:
Location header improvements(#1475)

* Create location header only on primary key columns, closes https://github.com/PostgREST/postgrest/issues/1461

* Return location header when pk columns not selected, fixes https://github.com/PostgREST/postgrest/issues/1162

steve-chavez committed:
refactor: move loadDbUriFile/SecretFile to Config

* make schema cacher filler use Appconfig
* change configRoleClaimKey to Either Text JSPath

steve-chavez committed:
refactor: move parseSecret out of App.postgrest

parseSecret only needs to be computed once, after the config is read.

steve-chavez committed:
refactor: move logStdout/corsPolicy to Middleware

* remove configQuiet from Config

configQuiet was not an end user setting.
The logging setup is now an internal parameter.

* move proxy uri validation to Private dir

steve-chavez committed:
Allow config file reloading with SIGUSR2

* move config validation to Config.hs
* Add tests for jwt-secret/app.settings.*/db-schema reload

Commit:
Bump nixpkgs pin to unstable as of 2020-07-13

wolfgangwalther committed:
Additionally allow "bearer" instead of only "Bearer" in Authorization… (#1558)

steve-chavez committed:
Schema cache reload with zero downtime (#1559)

* Improve error messages and comments

* Reorder Main.hs functions

wolfgangwalther committed:
Return 405 Method not Allowed for GET of volatile RPC instead of 500 (#1560)

steve-chavez committed:
refactor: add Hasql arrayColumn decoder

steve-chavez committed:
Fix charset=utf-8 appending to binary output

robx committed:
Allow base64-bytestring 1.2 and optparse-applicative 0.16

robx committed:
Make cabals bound more consistent

Specifically, allow point-updates to hasql-notifications, and
give a reasonable lower bound for network (the version that comess
with GHC 8.0 on stackage).

monacoremo committed:
Upgrade nixpkgs to current master and add an upgrade checklist (#1579)

* simplify haskell packages overrides based on new nixpkgs
* update Nix gitignore
* simplify Haskell packages overlay

Co-authored-by: monacoremo <monacoremo>

steve-chavez committed:
refactor: reorder PgVersion(95/96)Spec tests

* move json ops tests to JsonOperatorSpec
* move phfts operator tests to QuerySpec and RpcSpec
* add custom spec for pre-request header guc tests

robx committed:
Allow wai-extra 3.1

felixonmars committed:
Allow hspec-wai/hspec-wai-json 0.11 (#1595)

steve-chavez committed:
refactor: Change SET LOCAL gucs to set_config

wolfgangwalther committed:
Fix error messages on connection failure for postgres on localized Windows

wolfgangwalther committed:
add tests to pass regular arrays properly to RPCs

wolfgangwalther committed:
refactor moving findProc into ApiRequest to allow parsing parameters differently by proc

Commit:
Upgrade nixpkgs and add Postgres 13 support

wolfgangwalther committed:
Allow calling variadic functions with repeated query params or JSON array in body

steve-chavez committed:
Add log-level config

steve-chavez committed:
Add error log level

robx committed:
Allow wai-middleware-static 0.9

- Only serve static files on HEAD or GET requests.

monacoremo committed:
Use Postgres 13 in tests/docker-compose.yml

steve-chavez committed:
Add warn log level

steve-chavez committed:
Change default log-level, from info to error

wolfgangwalther committed:
Fix RPC return type handling for domains with composite base type

steve-chavez committed:
refactor: untangle addFiltersOrdersRanges

Remove convoluted function composition

wolfgangwalther committed:
Refactor tests: renaming json table to prevent hiding of native json type

wolfgangwalther committed:
Fix overloading of functions with unnamed arguments (specifically for prefer params=single-object)

wolfgangwalther committed:
Refactor tests: replace str with json QuasiQuoter where appropriate

wolfgangwalther committed:
Drop support for postgres 9.4

wolfgangwalther committed:
Update stack.yaml.lock (#1616)

ulidtko reviewed (commented):
LGTM in general, but two points:

 * update to a tagged version (7.0.1) would be more sane than pulling master
 * there're conflicts, what's up with that?

monacoremo committed:
Simplify Nix CI jobs

wolfgangwalther committed:
fix appveyour build script

wolfgangwalther committed:
fix build error on windows

wolfgangwalther committed:
fix travis build timeout swallowing error code

wolfgangwalther committed:
improve cirrus build time by properly using cabal cache

steve-chavez committed:
perf: change Text queries to ByteString

Improves performance by not utf8 encoding the whole query with
encodeUtf8. Only certain parts.

It's also a gradual step needed to use the Snippet type
from hasql-dynamic-statements.

wolfgangwalther committed:
correct issue numbers in changelog

wolfgangwalther committed:
ci: run spec tests against all pg versions, even if one of them fails (#1628)

- when the build fails no tests are run(circleci-agent step halt)
- when the build succeeds all tests are run, even if some of them fail(when: always)

wolfgangwalther committed:
ci: add coveralls.io integration

wolfgangwalther committed:
ci: remove io tests from stack-test because of random segmentation faults (#1627)

io tests already run on nix-build-test

steve-chavez committed:
refactor: Use hasql-dynamic on RPC/POST/PUT/PATCH

Change callProc/createWriteStatement to H.Snippet.

Parametrize the inputs on the same SQLFragments by taking advantage
of hasql-dynamic-statements. It's not necessary to parametrize
every input, inlining with pgFmtLit can still be used for queries
that can't be parametrized(like SET LOCALs).

Also adds hasql-dynamic-statements to Nix and stack.

wolfgangwalther committed:
fix: fix embedding through views with subqueries inside function calls

resolves #1608

Refactors the pfkSourceColumns query to use an intermediate JSON format
for parsing. This allows much more robust extraction of source columns.

steve-chavez committed:
perf: enable prepared statements for GET

- Parametrize filters
- Parametrize LIMIT/OFFSET
- Parametrize JSON path(select=col->$1)
- Single parameter for IN(use ANY)
- Also enable prepared statement for the EXPLAIN
  used on the estimated count.

steve-chavez committed:
perf: shortcut for proc with no variadic arg

Also refactor how rpc params are converted to json

wolfgangwalther committed:
fix non-variadic repeated param in variadic function

wolfgangwalther committed:
run stylish-haskell v0.12.1.0

wolfgangwalther committed:
refactor: split InsertSpec into InsertSpec and UpdateSpec

wolfgangwalther committed:
Add test to insert into a view and expect location header return

steve-chavez committed:
nix: add postgrest-docker-login wrapper func

steve-chavez committed:
circleci: make release job a machine

setup_remote_docker is somehow not working and gives:

docker: command not found

steve-chavez committed:
nix: fix postgrest-release-dockerhubdescription

steve-chavez committed:
appveyor: add nightly release

steve-chavez committed:
travisci: add nightly release

steve-chavez committed:
nix: change release scripts to use a version arg

steve-chavez committed:
circleci: add nightly release

For github and dockerhub

steve-chavez committed:
cirrusci: add nightly release

steve-chavez committed:
ci: add git sha with hours/minutes to nightly

steve-chavez committed:
ci: add nightly version to cabal file

steve-chavez committed:
Update BACKERS.md

steve-chavez committed:
circleci: fix nightly release

Also delete unused variable in release/default.nix

wolfgangwalther committed:
Enable embedding through multiple layers of views recursively (#1625)

* Include hidden views from the search path. Hidden views are views in unexposed schemas that are part of a view dependency chain.

* Change allSourceColumns to only return pk and fk columns

goteguru committed:
Removing single column limit from join table M2M mapping detection. (#1593)

wolfgangwalther committed:
fix: fix output of RPCs returning scalar values with multiple rows

resolves #1584

BREAKING CHANGE:
Changed output of RPCs to match return type better:
* single scalar return: value (unchanged)
* setof scalar return: array of values (new; was array of objects)
* single composite return: object (new; was array of objects)
* setof composite return: array of objects (unchanged)
* void return: nothing (new; was "null")

A single OUT column is now treated as "composite" instead of "scalar",
i.e. consistent with multiple OUT columns.

wolfgangwalther committed:
Remove executable bit on .nix and .sql file

wolfgangwalther committed:
Split StructureSpec into OpenApiSpec and OptionsSpec

wolfgangwalther committed:
Add support for Prefer tx=rollback

yvmarques committed:
Update README.md

Added "Since" for the security row level PostgreSQL has it since 9.5

wolfgangwalther committed:
Add test to ignore functions unnamed arguments for rpc disambiguation, resolves #1638

wolfgangwalther committed:
Add docker files to run nix dev environment on Windows

wolfgangwalther committed:
Make nix the one and only dev environment; remove obsolete tooling

wolfgangwalther committed:
Clean up postgrest --help output

steve-chavez committed:
Add db-prepared-statements config

wolfgangwalther committed:
Refactor tx-... config options to single tx-end option

steve-chavez committed:
cirrusci: add build timeout

wolfgangwalther committed:
added: cli option --dump-config prints loaded config and exits

This is most useful for automated tests for upcoming configuration
features. Can also be used for debugging.

steve-chavez committed:
Prepared statement for set_config

monacoremo committed:
Improve dev tools in the Nix environment (#1666)

* Improve dev tools in the nix environment

Added postgrest-build, postgrest-run, postgrest-clean, postgrest-check
and postgrest-watch tools. Changed order of postgrest-test-spec-all to
run backwards from pg13 to pg9.5. Added shellcheck to lint shell scripts
in nix environment.

Co-authored-by: monacoremo <monacoremo>

wolfgangwalther committed:
Use cabal v2-exec to run io tests, include io tests in nix-shell by default; remove obsolete ncat dependency (#1673)

Use cabal v2-exec to run io tests, include io tests in nix-shell by default; remove obsolete ncat dependency

Commit:
add autocompletion for postgrest-watch

wolfgangwalther committed:
improve error output for io-tests and with_tmp_db

wolfgangwalther committed:
Make transaction-rollback=true the default for test-suite (#1663)

Change test-suite to use db-tx-rollback-all = true by default

wolfgangwalther committed:
dev: improve postgrest-watch experience

* speed up io tests
* avoid infinite loop while watching io tests
* improve auto-complete with postgrest-watch postgrest-run
* detect when files are added or deleted

wolfgangwalther committed:
dev: add colors to io tests

wolfgangwalther committed:
feat: renamed config options with prefixes; added aliases for old names

* secret-is-base64 -> jwt-secret-is-base64
* role-claim-key -> jwt-role-claim-key
* max-rows -> db-max-rows
* pre-request -> db-pre-request
* root-spec -> db-root-spec
* db-schema -> db-schemas

This is not a breaking change, because aliases are added as well.

refactor: sorted all config keys alphabetically where applicable

wolfgangwalther committed:
fix: implement robust parsing of boolean config values

resolves #1572

wolfgangwalther committed:
feat: Read config directly from environment variables

resolves #1624

wolfgangwalther committed:
ci: fix cirrus low memory error

steve-chavez committed:
fix: retry connection on failed schema cache load (#1685)

Retry the connection when the
"Failed to load the schema cache" error happens.

Also die if the schema cache query has a syntax error.

wolfgangwalther committed:
run spec tests in parallel

reduces time to run postgrest-test-spec-all by 20-25%

monacoremo committed:
Migrate io-tests from bash to pytest

Authored-by: monacoremo <monacoremo>

wolfgangwalther committed:
ci: skip postgrest-release-dockerhubdescription for nightly builds

wolfgangwalther committed:
Use environment variables for configuration in developer tooling

Avoid the need for configuration files for temporary database connections.

Commit:
add test for spec idempotence

Commit:
upgrade nixpkgs and apply related fixes

Commit:
catch ReadTimeout in IO tests

monacoremo committed:
feat: Added --dump-schema CLI option to dump JSON of dbStructure schema cache.

Added postgrest-dump-schema to nix-shell returning a YAML dump of the test fixtures.

Authored-by: monacoremo <monacoremo>

monacoremo committed:
update stack to lts-16.26

wolfgangwalther committed:
feat: Add --example cli option to show example config file

wolfgangwalther committed:
nix: add "cd $rootdir" to all shell scripts via checked-shell-script

wolfgangwalther committed:
ci: remove broken travis coverage job

monacoremo committed:
improve io-tests

Tests are now run through unix socket by default allowing
parallel execution. Test setup configuration is done via
environment variables instead of config files.

monacoremo committed:
Add docs to all checked shell scripts in nix

wolfgangwalther committed:
Fix "unused-imports" warning for windows build

wolfgangwalther committed:
Fix spec test concurrency issue related to tx=commit on simple_pk

wolfgangwalther committed:
Add postgrest-coverage to show and upload hpc reports to codecov

wolfgangwalther committed:
Fix io tests not collecting coverage data when using run()

Before, postgrest was killed after each test and no coverage data was saved.
Now the process is terminated gracefully via SIGTERM first to allow writing the .tix file.

wolfgangwalther committed:
Fix postgrest-coverage throwing error when installed with nix-env

wolfgangwalther committed:
Add codecov to README.md
